Jenkinsfile.cd

```diff
@@ -92,6 +95,23 @@ def notifyingFailure() {
     }
 }
 
+def staticValidation() {
+    parallel([
+        'libindy'    : { runValdiationRust('libindy') },
+        'libnullpay' : { runValdiationRust('libnullpay') },
+        'libvcx'     : { runValdiationRust('vcx/libvcx') }
+    ])
+}
+
+def runValdiationRust(dir) {
+    node('ubuntu') {
+        setupRust()
+        sh "rustup component add clippy-preview"
+        sh "cd ${dir}"
+        sh "cargo clippy -- -W clippy::style -W clippy::correctness -W clippy::complexity -W clippy::perf"
+    }
+}
```

Comment:
you should use docker here
```
    node('ubuntu') {
        checkout scm
        def testEnv = buildTestEnv("ci/ubuntu.dockerfile ci", "Ubuntu")

        testEnv.inside {
            setupRust()
            sh "rustup component add clippy-preview"
            sh "cd ${dir} && cargo clippy -- -W clippy::style -W clippy::correctness -W clippy::complexity -W clippy::perf"
        }
    }
```

Reply:
What's the advantage to using docker here? Having to build the dockerfile adds a significant amount of time to the whole process. 
I understand the logic of avoiding installing clippy on the native executors, but I'm not sure the trade-off is worth it.

Reply:
There is no rust on native executors.
Building the dockerfile takes a long time only for the first time or when dockerfile has been changed.  The cache will be used for next buildings. We use common dockerfile for testing on ubuntu. So, using docker file doesn't affect our CI significantly.